erpnext/accounts/deferred_revenue.py

```diff
@@ -152,12 +165,24 @@ def calculate_amount(doc, item, last_gl_entry, total_days, total_booking_days, a
 			group by voucher_detail_no
 		'''.format(total_credit_debit, total_credit_debit_currency),
 			(doc.company, item.get(deferred_account), doc.doctype, doc.name, item.name), as_dict=True)
+
+		journal_entry_details = frappe.db.sql('''
+			SELECT sum(c.{0}) as total_credit, sum(c.{1}) as total_credit_in_account_currency, reference_detail_no
+			FROM `tabJournal Entry` p , `tabJournal Entry Account` c WHERE p.name = c.parent and
+			p.company = %s and c.account=%s and c.reference_type=%s and c.reference_name=%s and c.reference_detail_no=%s
+			and p.docstatus < 2 group by reference_detail_no
+		'''.format(total_credit_debit, total_credit_debit_currency),
+			(doc.company, item.get(deferred_account), doc.doctype, doc.name, item.name), as_dict=True)
+
 		already_booked_amount = gl_entries_details[0].total_credit if gl_entries_details else 0
+		already_booked_amount += journal_entry_details[0].total_credit if journal_entry_details else 0
```

Comment:
same entry will be added twice via GL and JE?

Reply:
I think it should only consider draft JE

Reply:
> same entry will be added twice via GL and JE?

@nabinhait 
While getting the already booked amount via GLE , the GLE's having voucher type only as 'Sales Invoice or Purchase Invoice' will be considered so same entry should not be added twice. Entries booked via Journal Entries will not be considered. Thats why I am getting both Draft and Submitted entries while fetching the amount from Journal Entry